BraveSharedTests/NSURLExtensionsTests.swift

```diff
@@ -195,6 +236,7 @@ class NSURLExtensionsTests: XCTestCase {
     let url = "http://[::1]/foo/bar".asURL!
     XCTAssertTrue(url.isIPv6)
     XCTAssertNil(url.baseDomain)
+    XCTAssertNil("[::1]".baseDomain)
     XCTAssertEqual(url.normalizedHost()!, "[::1]")
   }
 
```

Comment:
Shouldn't we add a different `port` check as well?

Reply:
> Shouldn't we add a different `port` check as well?

This tests domain. It doesn't include port so even if I put a port it'll be the same. Also the Chromium tests already handle all of the other cases. I'm only testing that our code works the same. The chromium code tests the APIs already :)